server/pkg/thread/thread.go

```diff
@@ -22,3 +27,73 @@ func (th *Thread) Comments() []*Comment {
 	}
 	return slices.Clone(th.comments)
 }
+
+func (th *Thread) HasComment(cid CommentID) bool {
+	if th == nil {
+		return false
+	}
+	return lo.SomeBy(th.comments, func(c *Comment) bool { return c.ID() == cid })
+}
+
+func (th *Thread) AddComment(c *Comment) error {
+	if th.comments == nil {
+		th.comments = []*Comment{}
+	}
+	if th.HasComment(c.ID()) {
+		return ErrCommentAlreadyExist
+	}
+
+	th.comments = append(th.comments, c)
+	return nil
+}
+
+func (th *Thread) UpdateComment(cid id.CommentID, content string) error {
+	if !th.HasComment(cid) {
+		return ErrCommentDoesNotExist
+	}
+
+	i := slices.IndexFunc(th.comments, func(c *Comment) bool { return c.ID() == cid })
+	th.comments[i].SetContent(content)
+	return nil
```

Comment:
```suggestion
	c, _ := lo.Find(th.comments, func(c *Comment) bool { return c.ID() == cid })
	if c == nil {
		return ErrCommentDoesNotExist
	}

	c.SetContent(content)
	return nil
```

When HasComment is used in conjunction with HasComment, two loops will be executed. With a slight code change, you can make one loop.

Reply:
Done!